src/Init/Data/Array/Basic.lean

```diff
@@ -59,6 +59,8 @@ def uget (a : @& Array α) (i : USize) (h : i.toNat < a.size) : α :=
 
 instance : GetElem (Array α) USize α fun xs i => i.toNat < xs.size where
   getElem xs i h := xs.uget i h
+  getElem? xs i  := if h : i.toNat < xs.size then some (xs.uget i h) else none
```

Comment:
Instead of copying the decidable default to each instance, can we have a helper (e.g., `decidableGetElem?`) defined in `GetElem` that is used in these instances?

Reply:
The problem is that the right hand side of `getElem` would still need to be repeated inside the call the `decidableGetElem?`.

I think we can perhaps resolve these default field difficulties by splitting `GetElem` in `GetElemBase` (with just `getElem), and extending it. I'll try.

Reply:
Works nicely, I think.

Reply:
@semorrison Looks good! Given that `GetElemBase` defines `getElem`, I think it would be best to give it the `GetElem` name and name the other something like `OptGetElem`, `GetElem?`, or `GetElem?!`.